cmd/earthly/build_cmd.go

```diff
@@ -296,8 +297,30 @@ func (app *earthlyApp) actionBuildImp(cliCtx *cli.Context, flagArgs, nonFlagArgs
 			return errors.Wrapf(err, "read %s", app.envFile)
 		}
 	}
+	validEnvNames := cliutil.GetValidEnvNames(app.cliApp)
+	for k := range dotEnvMap {
+		if _, found := validEnvNames[k]; !found {
+			app.console.Warnf("unexpected env \"%s\": as of v0.7.0, --build-arg values must be defined in .arg (and --secret values in .secret)", k)
+		}
+	}
+
+	argMap, err := godotenv.Read(app.argFile)
+	if err != nil {
+		// ignore ErrNotExist when using default .env file
+		if app.argFile != defaultArgFile || !errors.Is(err, os.ErrNotExist) {
+			return errors.Wrapf(err, "read %s", app.argFile)
+		}
+	}
+
+	secretsFileMap, err := godotenv.Read(app.secretFile)
+	if err != nil {
+		// ignore ErrNotExist when using default .env file
+		if app.secretFile != defaultSecretFile || !errors.Is(err, os.ErrNotExist) {
```

Comment:
Same comment as `argFile` about differentiating between the user explicitly passing the default file and us falling back to the default file.